### ISSUE TITLE
[research] AERA-v26.9 micro-op attribution after #687/#696

### ISSUE THREAD

**Issue by vinceackermann2-sys** [vinceackermann2-sys/tam-research#697]
## Purpose

Continue the frozen AERA-v26.9 systems investigation after the authoritative #687 stage-internal diagnostic completed successfully through #693/#696 but returned `next_target=None` because the dominant category differs by frozen system batch.

Authoritative current main at preregistration: `2b89fd0e6d12ad3f0d2d0ca870d838669cd7fddd`, tree `ad75aae06273fe42769855f5273f5e516a1dd20f`.

Authoritative source evidence: #696 comment `5569795957`, produced by run `34115412471`, job `101720942690`, attempt 1 on NVIDIA L4.

Frozen #687 findings:
- batch 8: full-call median `41.13203239440918` ms; stage-compute-ex-FICEM `26.655376447364688` ms; dominant `residual_stage_glue` at `6.067023945972323` ms; sparse experts `3.601471960544586` ms; latent reasoner `5.402431979775429` ms;
- batch 64: full-call median `76.7083511352539` ms; stage-compute-ex-FICEM `39.28654378838837` ms; dominant `sparse_experts` at `13.073423981666565` ms; residual stage glue `6.752576315775514` ms; latent reasoner `6.021375969052315` ms;
- `next_target=None`;
- source systems decision remains `FAIL_FROZEN_E2E_SYSTEMS_GATE`;
- no optimization authority was created.

## Question

Identify whether one lower-level operation family is responsible for a material share of both the small-batch residual/glue regime and the large-batch sparse-expert regime, or whether the architecture has genuinely separate overhead-bound and compute-bound bottlenecks that require separately preregistered optimization candidates.

## Boundary

Diagnostic-only. Authorize ZERO optimization, ZERO systems-PASS credit, and ZERO higher-stage authority.

Do not change learned equations, learned parameters, checkpoint bytes, routing decisions, sparse expert/depth decisions, FICEM equations/backend, write budget, durable state, throughput thresholds, frozen batch sizes `(8,64)`, token seed rule, or scientific training seed.

The diagnostic must execute the exact frozen v26.9 candidate on NVIDIA L4 with hard sparse routing and `update_memory=True`. It may instrument existing execution only; it must not skip, substitute, fuse, reorder, or otherwise optimize model operations.

At minimum decompose the two unresolved #687 regions into lower-level operation families sufficient to distinguish:

### Residual stage glue
- residual tensor adds and elementwise multiply/clamp/sigmoid work outside already-attributed children;
- slicing/view/reshape/expand/contiguous/indexing/mask construction and broadcast glue;
- concatenation/stack/state/tuple assembly and dtype restoration/transport glue;
- gather/index/select/scatter/reduction operations outside FICEM and outside the expert-bank child scope;
- any remaining unclassified stage-local CUDA work.

### Sparse experts
- routing probability/count work (`softmax`, `topk`, `argmax`, gate/weight normalization);
- selected expert weight gather/indexing;
- first expert projection/einsum or equivalent GEMM family;
- activation (`GELU`);
- second expert projection/einsum or equivalent GEMM family;
- weighted expert reduction;
- any remaining expert-local CUDA work.

Prefer operation-level profiling/instrumentation that preserves the frozen Python/model implementation rather than a copied/reimplemented forward. Any instrumentation overhead must be explicitly reported and this diagnostic may not be used as comparative systems evidence.

Use a fresh result path under this issue number. Preserve the frozen #687 result at `/vol/aera-v26/issue687-stage-internal-throughput-attribution/result.json` read-only as the parent evidence.

## Implementation governance

After this issue receives its number, freeze exact source main/tree, parent evidence comment/run/job/attempt, result path, source/checkpoint/model blobs, launcher/workflow/test blobs, branch, tree and commit before any branch/ref move.

Prefer exactly three additive issue-specific files:
1. one Modal L4 diagnostic launcher with zero-GPU `preflight` and one separately gated L4 diagnostic function;
2. one owner-only `issues: [opened]` workflow with fresh disjoint preauth/L4 prefixes;
3. one CPU/static contract test.

The Modal image must package every root-level dependency required remotely; #692's missing-root-import failure must not recur. CPU/static tests must explicitly verify the remote import graph and all no-authority/no-optimization invariants.

Full-repo CPU CI must pass on exactly one immutable PR head before merge. Any Actions attempt is consumed once and must never be rerun/retried/redispatched. After merge, require a fresh exact-main/tree/blob/result-absence audit, exactly one zero-GPU preauthorization trigger, then a separate exact-main L4 authorization and exactly one canonical L4 trigger.

## Decision rule

The result may name one `next_target` only if a single lower-level operation family is both:
1. the largest materially actionable contributor when the two frozen batch regimes are considered together, and
2. demonstrably relevant to both batch 8 and batch 64 rather than being dominant in only one regime.

If no such common family exists, return `next_target=None` and explicitly rank two regime-specific targets instead. That outcome authorizes no optimization automatically; a separately preregistered successor must choose and justify any optimization boundary.

No architecture freeze, S2, fresh scientific seed, independent replication credit, 100M scaling, or breakthrough claim is authorized here.

**Comment by vinceackermann2-sys** [comment]
## #697 pre-implementation micro-op attribution freeze

SOURCE_MAIN=2b89fd0e6d12ad3f0d2d0ca870d838669cd7fddd
SOURCE_TREE=ad75aae06273fe42769855f5273f5e516a1dd20f
BRANCH=research/aera-v26-9-issue697-micro-op-attribution
PREAUTH_PREFIX=[aera-v26-9-issue697-micro-op-preauth]
L4_PREFIX=[aera-v26-9-issue697-micro-op-l4]
RESULT_PATH=/vol/aera-v26/issue697-micro-op-attribution/result.json
PARENT_RESULT_PATH=/vol/aera-v26/issue687-stage-internal-throughput-attribution/result.json
PARENT_TRIGGER=696
PARENT_EVIDENCE_COMMENT=5569795957
PARENT_RUN=34115412471
PARENT_JOB=101720942690
PARENT_ATTEMPT=1
ISSUE687_LAUNCHER_BLOB=b9950c032686a496c6d944c30c441428499cb367
ISSUE665_LAUNCHER_BLOB=72f27391ff2f0a7bff8d4532f307ddc4869cf494
SCIENTIFIC_ADAPTER_BLOB=512572340cc09e2e7ad6729712258c12cb377ef2
RUNTIME_INTERFACE_BLOB=268644ac4edee15a4cc4e29d3fed7f61eeb3caa7
STAGE_V25_1_BLOB=1c3456d8040455b4cd1194db4c8586f77d0f3e43
TOKENWISE_V19_BLOB=98008bceb8c68af3bc346e5dfcc7a8218875661e
BASE_CORE_BLOB=ffe0341829fb905e40ef9ae3544f797471f1aa9c
NOHOST_V25_1_BLOB=237e5615cf32f644e8675808a6b3e9adaf04fb23
V26_9_BACKEND_BLOB=b81cc209f5d95abbe1fb8bd620c78e87c067bc19
AERA_CHECKPOINT_SHA256=f8aa92421801e8f190247e420632be5f0c20bc5ea8bf6bdeefe06686b3a31b30
TRANSFORMER_CHECKPOINT_SHA256=cdd5cab4439a709468d6607d45d82081b33e876b2e40d91d4a38ba139b219dd7
SYSTEM_BATCH_SIZES=8,64
TOKEN_SEED_RULE=138471 + 10000 + batch_size
MATERIAL_SHARE_MIN=0.15
MATERIAL_MS_MIN=1.0
LAUNCHER_BLOB=c82affc01d1ecb096371e8865dbd7b18e739f5e1
WORKFLOW_BLOB=d99ee80ebd31bcd4a032f13c5869d624a0c13d07
CPU_TEST_BLOB=49e640e254cfe596b17436073154083682a061e7
FROZEN_TREE=87a22d981caf77263574503749757ba77290cc9a
FROZEN_COMMIT=6719d66aab3b192cd96ee84d1d7bd63d1eaf31b6

Frozen detached implementation consists of exactly three additive issue-specific files. It changes no existing scientific/runtime/model/checkpoint/evaluator file. The profiler is diagnostic-only and cannot provide comparative systems evidence. No Modal execution, GPU use, optimization, systems PASS, architecture freeze, S2, fresh scientific seed, replication, 100M scaling, or breakthrough authority is created by this freeze.

**Comment by vinceackermann2-sys** [comment]
## #697 sole L4 micro-op attribution authorization

Authorize main: `6fad37d0e4c611fcaff480899b25dc190e9e9127`

Prerequisite evidence: sole #697 zero-GPU preauthorization trigger #699 completed successfully in Actions run `34124910097`, job `101751223863`, attempt `1`, with evidence comment `5571012795`; fresh result path `/vol/aera-v26/issue697-micro-op-attribution/result.json` was absent and zero GPU/model/new measurement/higher-stage authority was used.

Authorize exactly one canonical trigger under `[aera-v26-9-issue697-micro-op-l4]` to execute the frozen #697 diagnostic launcher/blob `c82affc01d1ecb096371e8865dbd7b18e739f5e1` on one NVIDIA L4, using the exact merged tree `87a22d981caf77263574503749757ba77290cc9a`, parent #687/#696 evidence, frozen checkpoints/blobs, batch sizes `(8,64)`, token seed rule, and frozen materiality thresholds.

This authorization is diagnostic-only. It authorizes no optimization, systems PASS, architecture freeze, S2, fresh scientific seed, independent replication credit, 100M scaling, or breakthrough claim. The sole L4 trigger and its Actions attempt are consumed regardless of outcome and must never be rerun/retried/redispatched.
